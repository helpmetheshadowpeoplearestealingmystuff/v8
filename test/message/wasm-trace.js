// Copyright 2020 the V8 project authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

// Flags: --trace-wasm --no-wasm-tier-up --liftoff --no-stress-opt

load('test/mjsunit/wasm/wasm-module-builder.js');

let builder = new WasmModuleBuilder();
let kRet23Function = builder.addFunction('ret_23', kSig_i_v)
                         .addBody([kExprI32Const, 23])
                         .exportFunc()
                         .index;
let kRet57Function = builder.addFunction('ret_57', kSig_l_v)
                         .addBody([kExprI64Const, 57])
                         .exportFunc()
                         .index;
let kRet0Function = builder.addFunction('ret_0', kSig_f_v)
                        .addBody(wasmF32Const(0))
                        .exportFunc()
                        .index;
let kRet1Function = builder.addFunction('ret_1', kSig_d_v)
                        .addBody(wasmF64Const(1))
                        .exportFunc()
                        .index;
builder.addFunction('main', kSig_v_v)
    .addBody([
      kExprCallFunction, kRet23Function, kExprDrop,  // -
      kExprCallFunction, kRet57Function, kExprDrop,  // -
      kExprCallFunction, kRet0Function, kExprDrop,   // -
      kExprCallFunction, kRet1Function, kExprDrop    // -
    ])
    .exportAs('main');

let instance = builder.instantiate();
instance.exports.main();
