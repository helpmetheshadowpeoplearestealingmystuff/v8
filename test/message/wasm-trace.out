Enter function
Enter function
Exit function -> 23
Enter function
Exit function -> 57
Enter function
Exit function -> 0.000000
Enter function
Exit function -> 1.000000
Exit function
